### ISSUE TITLE
add support for other call lookup sources?

### ISSUE THREAD

**Issue by classabbyamp** [miaowware/qrm2#352]
should there be a fallback order for `?call`, like:

1. qrz
2. calllook
3. ???
4. disabled

**Comment by artiemog** [comment]
[HamQTH](https://www.hamqth.com/) would be another good potential data source, if you're thinking about adding them.

**Comment by classabbyamp** [comment]
TODO with this change: remove deprecated `?qrz` alias

**Comment by classabbyamp** [comment]
- callook has [json and xml apis](https://callook.info/api_reference.php)
    - no login needed
- hamqth has [xml api](https://www.hamqth.com/developers.php#xml_search)
    - requires login

**Comment by 0x5c** [comment]
Callook not needing a login is a massive plus
JSON too

**Comment by classabbyamp** [comment]
yeah, I think in the default config, the fallback order should be:

1. QRZ
2. HamQTH
3. Callook

this would allow people to choose the better data sources if they want to pay $$ (qrz) or make an account (hamqth), but still allow some lookup via callook with no creds.

not sure if we should do a fallback on callsign not found, or just on credentials not found. thoughts?

**Comment by 0x5c** [comment]
Both, since someone may have qrz but not hamqth, requiring callook for every notfounds on qrz

Might be good idea to add commands (or flags) to directly look on a specific data source

**Comment by classabbyamp** [comment]
also qrzcq [docs](https://www.qrzcq.com/docs/api/xml/)

**Comment by LFGSaito** [comment]
could we add examtool’s lookup? It’s a json output, and we were able to get it, to update the ULS database every 10 minutes so freshly granted callsigns are mostly in there. a example call is https://alpha.exam.tools/api/uls/lookup/wy4rc

**Comment by classabbyamp** [comment]
we'll consider it

**Comment by LFGSaito** [comment]
not really, but you can give it a FRN, callsign, or ULS license key. eg. https://exam.tools/api/uls/lookup/kd7bbc and https://exam.tools/api/uls/lookup/0006800387 and https://exam.tools/api/uls/lookup/526662 all load the same thing

**Comment by classabbyamp** [comment]
what about documentation of what the response can contain or how it errors?

**Comment by 0x5c** [comment]
For many reasons, we wouldn't add this as a a direct lookup source, but we'd build a "is my call granted" type of command, so that people who freshly passed their exam/upgrade or changed their call could check if it's in the system.

Conditional on obtaining documentation on the API, we'd add this.

**Comment by 0x5c** [comment]
Moved the call check command to this new issue: #382
